dmoj/executors/RUST.py

```diff
@@ -56,15 +59,44 @@ def create_files(self, problem_id, source_code, *args, **kwargs):
         with open(self._file('Cargo.toml'), 'wb') as f:
             f.write(CARGO_TOML)
 
+    shared_target = None
+
+    @classmethod
+    def get_shared_target(cls):
+        if cls.shared_target is None:
+            cargo_dir = os.path.expanduser('~/.cargo')
+            collisions = 0
+            judge_id = env.get('id', 'anonymous')
+            hasher = hashlib.sha256()
+            hasher.update(judge_id.encode('utf-8'))
```

Comment:
After thinking about this more, by doing the flock thing we don't even need to include the hash in the name, just the collision count.

Reply:
Done.